### PR TITLE
api: add download analysis output files endpoint

### LINKED ISSUE #21: api: add download analysis output files endpoint
Implement endpoint `GET` `/api/analyses/:analysis_uuid/workspace/outputs/:file_path` following previous PRs style (see [here](https://github.com/reanahub/reana-server/pull/17/files))
- [x] Copy last `workflow-controller` OpenAPI spec.
- [x] Use bravado client to make the call. Remember to handle exceptions properly as described in https://github.com/reanahub/reana-server/issues/18.
- [x] Test creating some files under `<analysis_dir>/outputs/` and then download some using `curl` call direct to `server`.

### PR DESCRIPTION
**Pull request by diegodelemos**
* (closes #21).

### CODE REVIEW

File: docs/openapi.json [hunk 1]
```diff
@@ -417,6 +417,73 @@
...
+            "description": "Requests succeeded. The file has been downloaded.",
+            "schema": {
+              "type": "file"
+            }
+          },
+          "400": {
+            "description": "Request failed. The incoming data specification seems malformed."
+          },
+          "404": {
+            "description": "Request failed. User doesn't exist.",
```
Comment (tiborsimko) [inline docs/openapi.json]: Better `file_name` does not exist for output file handling endpoints...

File: reana_server/rest/analyses.py [hunk 2]
```diff
@@ -488,3 +491,94 @@ def set_analysis_status(analysis_id):  # noqa
...
+            file_name=file_name).result()
+
+        return send_file(
+            io.BytesIO(http_response.raw_bytes),
+            attachment_filename=file_name,
+            mimetype='multipart/form-data'), 200
+    except (KeyError, HTTPBadRequest) as e:
+        return jsonify({"message": str(e)}), 400
+    except HTTPError as e:
+        # we have this option, to blindly mirror responses from RWC or
```
Comment (diegodelemos) [inline reana_server/rest/analyses.py]: @tiborsimko @hjhsalo any opinion here? I would rather prefer to explicitly specify exceptions so you can directly check and update the `docstring`. If we just redirect what comes from RWC we would have to check RWC `openapi` spec to write the `docstring` (which is likely to not going to happen)

Comment (tiborsimko) [reply reana_server/rest/analyses.py]: I'd probably favour explicit over implicit...

File: reana_server/openapi_connections/reana_workflow_controller.json [hunk 3]
```diff
@@ -454,6 +454,223 @@
...
+        "produces": [
+          "multipart/form-data"
+        ],
+        "responses": {
+          "200": {
+            "description": "Requests succeeded. The list of input files has been retrieved.",
+            "schema": {
+              "items": {
+                "properties": {
+                  "last-modified": {
```
Comment (hjhsalo) [inline reana_server/openapi_connections/reana_workflow_controller.json]: Not to be added for this PR, but maybe checksumming (MD5, SHA1) would be a nice improvement at some point.

File: reana_server/rest/analyses.py [hunk 4]
```diff
@@ -20,9 +20,12 @@
 # submit itself to any jurisdiction.
 
 """Reana-Server analysis-functionality Flask-Blueprint."""
-from bravado.exception import HTTPForbidden, HTTPBadRequest, HTTPNotFound
+import io
+
+from bravado.exception import (HTTPBadRequest, HTTPError, HTTPForbidden,
```
Comment (hjhsalo) [inline reana_server/rest/analyses.py]: `HTTPError` doesn't seem to be used.

Comment (diegodelemos) [reply reana_server/rest/analyses.py]: indeed, thanks

**Pull request merged**